tests/codegen/gpu-kernel-abi.rs

```diff
@@ -0,0 +1,23 @@
+// Checks that the gpu-kernel calling convention correctly translates to LLVM calling conventions.
+
+//@ revisions: amdgpu nvptx
+//@ [amdgpu] compile-flags: --crate-type=rlib --target=amdgcn-amd-amdhsa -Ctarget-cpu=gfx900
+//@ [amdgpu] needs-llvm-components: amdgpu
+// amdgpu target is not yet merged
+//@ [amdgpu] should-fail
```

Comment:
this doesn't really work:
```
  
  thread 'main' panicked at src/tools/compiletest/src/header.rs:1559:17:
  missing LLVM component amdgpu, and COMPILETEST_REQUIRE_ALL_LLVM_COMPONENTS is set
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
  Build completed unsuccessfully in 0:37:29
    local time: Wed Jan 15 03:17:37 UTC 2025
    network time: Wed, 15 Jan 2025 03:17:37 GMT
```

Reply:
Oh, Rust’s testing is too advanced for this to go through 😃